osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/map.js

```diff
@@ -72,25 +65,29 @@
         this.selectElementExistingExcerpts.addEventListener('change', function() {
             // we have single select, if it would happen to have two selected, ignore it.
             var excerptOption = $( "select[id=id_existing_excerpts] option:selected")[0];
-            if(excerptOption.getAttribute('data-geometry') === 'boundingbox') {
-                var bounds = this.locationFilter.getBounds();
-                bounds._northEast.lat = excerptOption.getAttribute('data-north');
-                bounds._southWest.lng = excerptOption.getAttribute('data-west');
-                bounds._northEast.lng = excerptOption.getAttribute('data-east');
-                bounds._southWest.lat = excerptOption.getAttribute('data-south');
-                this.locationFilter.setBounds(bounds);
-                // enable excerpt on map for case it was disable before (e.g. by clicking on a not-boundingbox excerpt in the list)
-                this.locationFilter.enable();
-            } else {
-                // no existing excerpt of type boundingbox -> disable excerpt because we are not able to show not-boundingbox excerpts on map
-                this.locationFilter.disable();
-            }
+            this._setLocationFilterFromExcerptID(excerptOption.value);
         }.bind(this));
 
         // enable excerpt on map on change of form mode (existing excerpt or new excerpt)
         this.formElementPartsSwitcher.addEventListener('change', function(event) {
             this.locationFilter.enable();
         }.bind(this));
+
+        this._setLocationFilterFromExcerptID = function(ID) {
+            var that = this;
```

Comment:
Isn't it conventional to use a variable named `self` as alias for `this` for closure purposes?